internal/docker/client.go

```diff
@@ -48,10 +51,45 @@ func GetClient() (*client.Client, error) {
 		}
 	}
 
-	dockerClient, err := client.NewClientWithOpts(clientOpts...)
-	if err != nil {
-		return nil, fmt.Errorf("failed create docker client: %w", err)
+	// This tries to create a docker client with the default options
+	// If it fails, it tries to create a client with the runtime specific path
+	dockerClient, err := newClient("", clientOpts...)
+	if err == nil {
+		err := checkConnection(dockerClient)
+		if err == nil {
+			return dockerClient, nil // Successfully connected
+		}
+	}
+
+	// If the client socket didn't work, try the GOOS specific path
+	// This is useful for macOS, where the default Unix socket for newer distributions of docker desktop is different
+	dockerClient, err = newClient(runtime.GOOS, clientOpts...)
+	if err == nil {
+		err := checkConnection(dockerClient)
+		if err == nil {
+			return dockerClient, nil // Successfully connected
+		}
+	}
+
+	// If both attempts failed
+	return nil, fmt.Errorf("failed to connect to Docker daemon. Ensure Docker is running and accessible")
+}
+
+func newClient(os string, opts ...client.Opt) (*client.Client, error) {
```

Comment:
It surprised me that `os` changes between calls during the same invocation. Can we find a way to refactor so that this parameter never lies? (Passing `""` as `os` when we know perfectly well there is a value in runtime.GOOS feels like lying to the method.)

My suggestion to separate like this:
1. Get a []string of possible socket paths. This method has os specific behavior, and maybe accepts an `os` string parameter so it's easier to unit test.
2. Loop over them until one works. This method doesn't have os specific behavior.